engine/src/main/java/org/terasology/rendering/opengl/FrameBuffersManager.java

```diff
@@ -252,109 +344,123 @@ public void saveScreenshot() {
             }
         };
 
-        CoreRegistry.get(ThreadManager.class).submitTask("Write screenshot", task);
-
+        threadManager.submitTask("Write screenshot", task);
```

Comment:
The idea is that some classes are automatically instantiated, for example those who are annotated with `@RegisterSystem`. One way is to call a (default) constructor and fill all fields that are annotated with `@In` through reflection with variables from `CoreRegistry` or  or pass them to a constructor that takes them as parameters.

For now, I would add a field `threadManager` and manually fill it in the constructor to facilitate later refactoring, but that's just my opinion - you can as well just leave it as-is.